pennylane/tape/qscript.py

```diff
@@ -188,7 +188,7 @@ def __init__(
     def __repr__(self):
         return f"<{self.__class__.__name__}: wires={self.wires.tolist()}, params={self.num_params}>"
 
-    @property
+    @cached_property
```

Comment:
Seeing this makes me wonder whether we should be using `cached_property` in more places (throughout PL), where we manually create an attribute `_...` and compute it only upon the first call to the corresponding method/property. :thinking:

Reply:
Definitely!

It does rely on `__eq__` and `__hash__`, so if calculating those can get expensive (ie operator), it's probably worth still doing it manually.

Reply:
Perfect, thanks!